bambi/plots/effects.py

```diff
@@ -0,0 +1,349 @@
+    elif use_hdi:
+        y_hat_bounds = az.hdi(y_hat, hdi_prob)[f"{response_name}_{target}"].T
+    else:
+        lower_bound = round((1 - hdi_prob) / 2, 4)
+        upper_bound = 1 - lower_bound
+        y_hat_bounds = y_hat.quantile(q=(lower_bound, upper_bound), dim=("chain", "draw"))
+
+    cap_data["estimate"] = y_hat_mean
+    cap_data["hdi_3%"] = y_hat_bounds[0]
+    cap_data["hdi_97%"] = y_hat_bounds[1]
+
+    return cap_data
+
+
+@dataclass
+class Contrast:
+    term: Union[str, list]
+    value: Union[list, np.ndarray, None]
+
+
+@dataclass
+class Response:
+    name: str
+    target: str
+
+
+@dataclass
+class ContrastEstimate:
+    comparison: Dict[str, xr.DataArray]
+    hdi: Dict[str, xr.Dataset]
+
+
+def comparisons(
+    model: bmb.Model,
+    idata: az.InferenceData,
+    contrast: Union[str, dict, list],
+    conditional: Union[str, dict, list],
+    comparison_type: str = "diff",
+    target: str = "mean",
+    use_hdi: bool = True,
+    hdi_prob=None,
+    transforms=None,
+) -> pd.DataFrame:
+    """Compute Conditional Adjusted Comparisons
+
+    Parameters
+    ----------
+    model : bambi.Model
+        The model for which we want to plot the predictions.
+    idata : arviz.InferenceData
+        The InferenceData object that contains the samples from the posterior distribution of
+        the model.
+    contrast : str, dict, list
+        The predictor name whose contrast we would like to compare.
+    conditional : str, dict, list
+        The covariates we would like to condition on.
+    comparison_type : str, optional
+        The type of comparison to plot. Defaults to 'diff'.
+    target : str
+        Which model parameter to plot. Defaults to 'mean'. Passing a parameter into target only
+        works when pps is False as the target may not be available in the posterior predictive
+        distribution.
+    use_hdi : bool, optional
+        Whether to compute the highest density interval (defaults to True) or the quantiles.
+    hdi_prob : float, optional
+        The probability for the credibility intervals. Must be between 0 and 1. Defaults to 0.94.
+        Changing the global variable ``az.rcParam["stats.hdi_prob"]`` affects this default.
+    legend : bool, optional
+        Whether to automatically include a legend in the plot. Defaults to ``True``.
+    transforms : dict, optional
+        Transformations that are applied to each of the variables being plotted. The keys are the
+        name of the variables, and the values are functions to be applied. Defaults to ``None``.
+
+    Returns
+    -------
+    pandas.DataFrame
+        A dataframe with the comparison values, highest density interval, contrast name,
+        contrast value, and conditional values.
+
+    Raises
+    ------
+    ValueError
+        When ``hdi_prob`` is not > 0 and < 1.
+    """
+
+    if isinstance(contrast, dict):
+        contrast_name, contrast_level = next(iter(contrast.items()))
```

Comment:
If I understand correctly, when you have a dict like `{"a": 1, "b":2, "c": 3}, this sets `contrast_name` to `"a"` and `contrast_level` to `1`. Why do we keep only the key/value of the first pair?

Reply:
Also, it seems you only care about `contrast_name`? I'm not sure where `contrast_level` is used.

Reply:
perhaps `next(iter(contrast.keys()))` is better, and I would add a comment explaining why you do that

Reply:
Regarding:

_Comment 1_: Correct, and I only keep the key/value of the first pair since I have not added functionality for computing cross-contrasts (when a user wants to compare > 1 contrast predictor)

_Comment 2 & 3_: `contrast_level` is not used. Changed to use `.keys()`. Perhaps an error should be raised if they attempt to pass > 1 contrast predictor as this is not implemented.